dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -312,7 +312,7 @@ private static ArtifactCache loadArtifactInfo(List<Artifact> artifacts) {
 
     for (Artifact artifact : artifacts) {
       DependencyGraph completeDependencies =
-          dependencyGraphBuilder.buildFullDependencyGraph(ImmutableList.of(artifact));
+          dependencyGraphBuilder.buildVerboseDependencyGraph(artifact);
```

Comment:
This change does not affect Linkage Check column in the dashboard. If you want to change the Linkage Checker's class path, you need to update code around line 184:

```
  private static Path generate(Bom bom) throws IOException, TemplateException, URISyntaxException {

    ImmutableList<Artifact> managedDependencies = bom.getManagedDependencies();

    ClassPathResult classPathResult = classPathBuilder.resolve(managedDependencies);
    ImmutableList<ClassPathEntry> classpath = classPathResult.getClassPath();

    LinkageChecker linkageChecker = LinkageChecker.create(classpath);
...
```

Reply:
I just checked buildVerboseDependencyGraph for com.google.http-client:google-http-client:jar:1.35.0. It generates additional linkage errors for the missing classes in the optional dependencies.

<img width="1394" alt="Screen Shot 2020-06-23 at 14 44 03" src="https://user-images.githubusercontent.com/28604/85444216-1f030180-b560-11ea-9d41-1ab30514cae2.png">

Reply:
This is interesting. Two of those might be true positives but I need to investigate. I wonder how this class is guarding those usages?

Reply:
After looking into commons-logging I think:

1. Those are unlikely to come up in practice but are true positives.
2.  No one should depend on Commons-logging for multiple reasons, including that it does not properly guard usage of its optional dependencies.

This does add weight to your idea about distinguishing linkage errors caused by dependency conflicts from linkage errors caused by missing optional dependencies.